cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -251,6 +262,8 @@ public String commitAndPush(final CommitMeta commitMeta,
       }
     }
 
+    metricsContext.gauge(SourceControlManagement.METRIC_COMMIT_PUSH_LATENCY,
+        System.currentTimeMillis() - operationStartTime);
```

Comment:
nit:
There is some pro/con about using `currentmillis` vs `nanotime` (https://stackoverflow.com/questions/52946694/whats-a-clean-way-to-time-code-execution-in-java)
Do we use this pattern to capture operation timing in other places? In that case maybe we can have a higher order function for this

Reply:
Switched to using the StopWatch class from guvava. I don't see the use of higher order functions in CDAP. Here are a couple of latency metrics that I found in CDAP: 
* https://github.com/cdapio/cdap/blob/7ce4f2ae477312abbe93d058d1b3d256f09a8edf/cdap-tms/src/main/java/io/cdap/cdap/messaging/subscriber/AbstractMessagingPollingService.java#L209-L218
* https://github.com/cdapio/cdap/blob/360d7dbc4d79633e3ac9c8cc512884a9290b4c17/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java#L1462-L1465